src/transformStateWithClones.js

```diff
@@ -7,7 +7,45 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrayOfClones = [];
+  const stateClone = { ...state };
+
+  for (const action of actions) {
+    switch (true) {
+      case (action.type === 'addProperties'):
+        for (const key in action.extraData) {
+          stateClone[key] = action.extraData[key];
+        }
```

Comment:
This for loop is redundant. Use `Object.assign(stateClone, action.extraData);` instead